### PR TITLE
[regression-test](Nereids) add agg function, tvf, generator, window function test cases

### PR DESCRIPTION
**Pull request by sohardforaname**
# Proposed changes

Issue Number: close #xxx

## Problem summary

add agg_function, tvf, generator, window_function test for nereids and add more feature to gen.py

## Checklist(Required)

* [ ] Does it affect the original behavior
* [x] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: regression-test/suites/nereids_function_p0/window_function/window_fn.groovy [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    qt_sql_dense_rank "select ktint, ksint, dense_rank() over(partition by ktint order by ksint) as rank from fn_test"
+    qt_sql_dense_rank_notnull "select ktint, ksint, dense_rank() over(partition by ktint order by ksint) as rank from fn_test_not_nullable"
+
+    qt_sql_first_value "select ktint, ksint, first_value(kint) over(partition by ktint order by ksint) as fv from fn_test"
+    qt_sql_first_value_notnull "select ktint, ksint, first_value(kint) over(partition by ktint order by ksint) as fv from fn_test_not_nullable"
+
+    qt_sql_lag "select ktint, ksint, lag(kint, 2, 1) over(partition by ktint order by ksint) as lag from fn_test"
+    qt_sql_lag_notnull "select ktint, ksint, lag(kint, 2, 1) over(partition by ktint order by ksint) as lag from fn_test_not_nullable"
+
+    qt_sql_last_value "select ktint, ksint, last_value(kint) over(partition by ktint order by ksint) as lv from fn_test"
```
Comment (morrySnow) [inline regression-test/suites/nereids_function_p0/window_function/window_fn.groovy:31]: need enumerate with/without partition, with/without order by and all window frame

File: regression-test/suites/nereids_function_p0/agg_function/agg.groovy [hunk 2]
```diff
@@ -0,0 +1,597 @@
...
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+suite("nereids_agg_fn") {
+	sql 'use regression_test_nereids_function_p0'
+	sql 'set enable_nereids_planner=true'
+	sql 'set enable_fallback_to_original_planner=false'
+	qt_sql_any_value_AnyData_gb "select any_value(kint) from fn_test group by kbool order by kbool"
```
[the remark this answers is not in the document]
Comment (sohardforaname) [reply regression-test/suites/nereids_function_p0/agg_function/agg.groovy:22]: ok

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run clickbench p0

**Comment by sohardforaname** [comment]
run p0

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run p0

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run p0

**Comment by sohardforaname** [comment]
run p0

**Comment by sohardforaname** [comment]
run p0

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run p0

**Comment by sohardforaname** [comment]
run p0

**Comment by sohardforaname** [comment]
run p0

**Comment by sohardforaname** [comment]
run p0

**Comment by sohardforaname** [comment]
run p0

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run p0

**Comment by sohardforaname** [comment]
run p0

**Comment by sohardforaname** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**